fabbwled-backend/src/main/java/ch/bbw/fabbwled/lands/book/yaml/Action.java

```diff
@@ -1,18 +1,45 @@
 package ch.bbw.fabbwled.lands.book.yaml;
 
 import ch.bbw.fabbwled.lands.book.SectionId;
+import ch.bbw.fabbwled.lands.exception.FabledTechnicalException;
 
 import java.util.List;
 import java.util.Optional;
 
 public interface Action {
-    record TextAction(String text) implements Action {}
+    void simpleVerify();
 
-    record IfAction(Condition condition, List<Action> then, Optional<List<Action>> else_) implements Action {}
+    record TextAction(String text) implements Action {
+        @Override
+        public void simpleVerify() {}
+    }
+
+    record IfAction(Condition condition, List<Action> then, Optional<List<Action>> else_) implements Action {
+        @Override
+        public void simpleVerify() {
+            then.forEach(Action::simpleVerify);
+            else_.ifPresent(else_ -> else_.forEach(Action::simpleVerify));
+        }
+    }
 
-    record TurnToAction(SectionId sectionId) implements Action {}
+    record TurnToAction(SectionId sectionId) implements Action {
+        @Override
+        public void simpleVerify() {}
+    }
 
     record Choice(List<SingleChoice> choices) implements Action {
+        @Override
+        public void simpleVerify() {
+            if (choices.isEmpty()) {
+                throw new FabledTechnicalException("Found choice with no choices, this doesn't make sense.");
+            }
+            if (choices.size() == 1) {
```

Comment:
how about extending the json schema to have a strict definition of the `choice` key featuring
https://datatracker.ietf.org/doc/html/draft-fge-json-schema-validation-00#section-5.3.3 `minItems` ?

Reply:
sounds good too